File: frontend/src/microservices/traveler/objects/TravelerContactInfo.ts
// 本文件定义 traveler 模块的 `TravelerContactInfo`，作为联系信息传输数据并提供 JSON 编解码。

export type TravelerContactInfo = {
  phone: string
  email: string | null
}

export const travelerContactInfoFromJson = (json: string): TravelerContactInfo =>
  JSON.parse(json) as TravelerContactInfo

export const travelerContactInfoToJson = (value: TravelerContactInfo): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/traveler/objects/TravelerContactInfo.scala
// 本文件定义 traveler 模块的联系传输对象 `TravelerContactInfo`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TravelerContactInfo(
    phone: String,
    email: Option[String]
)

object TravelerContactInfo:
  given sourceEncoder: Encoder[TravelerContactInfo] = deriveEncoder
  given sourceDecoder: Decoder[TravelerContactInfo] = deriveDecoder
